frontend/src/scenes/project/Settings/CorrelationConfig.tsx

```diff
@@ -18,31 +22,103 @@ export function CorrelationConfig(): JSX.Element {
-                            {personProperties
-                                .map(({ name }) => name)
-                                .sort()
-                                .map((name) => (
-                                    <Select.Option key={name} value={name}>
-                                        {name}
-                                    </Select.Option>
-                                ))}
-                        </Select>
-                    </>
+                    <PersonPropertySelect
+                        onChange={handleChange}
+                        selectedProperties={currentTeam.correlation_config.excluded_person_property_names || []}
+                    />
                 )}
             </div>
         </div>
     )
 }
+
+interface PersonPropertySelectProps {
+    onChange: (names: string[]) => void
+    selectedProperties: string[]
+}
+
+const PersonPropertySelect = ({ onChange, selectedProperties }: PersonPropertySelectProps): JSX.Element => {
+    const { open, toggle, hide } = usePopup()
+
+    const handleChange = (name: string): void => {
+        onChange(Array.from(new Set(selectedProperties.concat([name]))))
+    }
+
+    const handleRemove = (name: string): void => {
+        onChange(selectedProperties.filter((p) => p !== name))
+    }
+
+    return (
+        <div className="mb">
+            {selectedProperties.length > 0 &&
+                selectedProperties.map((name) => {
+                    return (
+                        <Row
+                            key={name}
+                            align="middle"
+                            className="property-filter-row mt-05 mb-05"
+                            style={{
+                                width: '100%',
+                                margin: '0.25rem 0',
+                                padding: '0.25rem 0',
+                            }}
+                        >
+                            <Button type="primary" shape="round">
+                                {name}
+                            </Button>{' '}
+                            <CloseButton
+                                className="ml-1"
+                                onClick={() => handleRemove(name)}
+                                style={{ cursor: 'pointer', float: 'none', marginLeft: 5 }}
+                            />
+                        </Row>
+                    )
+                })}
+            <Row>
+                <Popup
+                    visible={open}
+                    overlay={
+                        <TaxonomicFilter
+                            onChange={(_, value) => {
+                                handleChange(value as string)
+                                hide()
+                            }}
+                            taxonomicGroupTypes={[TaxonomicFilterGroupType.PersonProperties]}
+                        />
+                    }
+                >
+                    {({ setRef }) => (
```

Comment:
Q: Why can't we just use the simple form here, without the `setRef` ? Doesn't seem to be used?

Reply:
Is it not used by Popup, i.e. [here](https://github.com/PostHog/posthog/blob/use-taxo-filter/frontend/src/lib/components/Popup/Popup.tsx#L43:L43)

We need to use setRef to have Popup attach events. I also initially just used antd Popover here, but it seems for consistency that Popup must be used, although I could be wrong on that.